nikola/nikola.py

```diff
@@ -1926,7 +1927,8 @@ def path(self, kind, name, lang=None, is_link=False, **kwargs):
             else:
                 return link
         else:
-            return os.path.join(*path)
+            # URLs should always use forward slash separators, even on Windows
+            return pathlib.PurePosixPath(*path)
```

Comment:
This should return a string, not a pathlib.Path instance.

Reply:
Ahar, fair point, thank you. I was curious about why this wasn't causing problems in the tests, so I had a quick look around and it seems like almost every call to this function is using the return value as a component passed to os.path.join(), which accepts pathlib.Paths. So that explains why the problem isn't visibly manifesting. But yes, you are right, future callers to this (and maybe the existing one in Nikola.link()) might barf on pathlib.Paths. So I'll convert it to str, and swallow my idealism over this change not requiring a test change.